src/entrypoint.sh

```diff
@@ -155,20 +169,45 @@ function git_init() {
     mkdir -p "${HOME}"/.ssh
     ssh-keyscan -t rsa "${source_repo_hostname}" >> "${HOME}"/.ssh/known_hosts
   else
-    info "the source repository is located within GitHub."               
-    if [[ -n "${SOURCE_GH_TOKEN}" ]]; then      
-      unset GITHUB_TOKEN      
-      gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${SOURCE_GH_TOKEN}"        
-      if [[ -n "${TARGET_GH_TOKEN}" ]]; then
-        gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${TARGET_GH_TOKEN}"
-      fi            
-      gh auth switch      
-      gh auth setup-git --hostname "${source_repo_hostname}"
-      info "done set git global configuration"    
-    else
-      info "default token to be used" 
-      gh auth setup-git --hostname "${source_repo_hostname}"      
-    fi       
+    info "the source repository is located within GitHub."
+  fi
+  echo "::endgroup::"
+}
+
+#######################################
+# doing the login to the source repository using gh cli
+# Arguments:
+#   source_repo_hostname
+#######################################
+function gh_login_src_github() {
+  echo "::group::login src github"
+  local source_repo_hostname=$1
+  # GITHUB_TOKEN is deprecated and can be removed in the future
+  if [[ -n "${SOURCE_GH_TOKEN}" ]] || [[ -n "${GITHUB_TOKEN}" ]] &>/dev/null; then
+    ################################
+    if [[ -n "${GITHUB_TOKEN}" ]] &>/dev/null; then
+      warn "github_token parameter is deprecated please use source_gh_token."
+      info "setting SOURCE_GH_TOKEN"
+      export SOURCE_GH_TOKEN="${GITHUB_TOKEN}"
+      unset GITHUB_TOKEN
+    fi
+    ###############################
+    if [[ -z "${SOURCE_GH_TOKEN}" ]] &>/dev/null; then
+      err "Missing input 'source_gh_token: \${{ secrets.GITHUB_TOKEN }}'.";
+      exit 1;
+    fi
+    info "source server url: ${source_repo_hostname}"
+    info "logging out"
+    gh auth logout --hostname "${source_repo_hostname}" || debug "not logged in"
```

Comment:
I don't think we need to log out @AndreasAugustin. We could do another login and do a gh auth switch action to switch accounts. What do you think about this?

Reply:
Actually I removed the gh auth switch because we do not have the user information.
We don't know if people actually have done further logins on the runner before. Therefore the `gh auth switch` possibly fails and also with an eye onto further development it is harder to keep track.

Out of a feeling I do not super like both options 😿